cmd/crank/beta/diff/diff.go

```diff
@@ -0,0 +1,178 @@
+
+	// Configuration options
+	NoColor bool          `help:"Disable colorized output."                name:"no-color"`
+	Compact bool          `help:"Show compact diffs with minimal context." name:"compact"`
+	Timeout time.Duration `default:"1m"                                    help:"How long to run before timing out."`
+	QPS     float32       `default:"0"                                     help:"Maximum QPS to the API server."`
+	Burst   int           `default:"0"                                     help:"Maximum burst for throttle."`
+}
+
+// Help returns help instructions for the diff command.
+func (c *Cmd) Help() string {
+	return `
+This command returns a diff of the in-cluster resources that would be modified if the provided Crossplane resources were applied.
+
+Similar to kubectl diff, it requires Crossplane to be operating in the live cluster found in your kubeconfig.
+
+Examples:
+  # Show the changes that would result from applying xr.yaml (via file) in the default 'crossplane-system' namespace.
+  crossplane diff xr.yaml
+
+  # Show the changes that would result from applying xr.yaml (via stdin) in the default 'crossplane-system' namespace.
+  cat xr.yaml | crossplane diff --
+
+  # Show the changes that would result from applying xr.yaml, xr1.yaml, and xr2.yaml in the default 'crossplane-system' namespace.
+  cat xr.yaml | crossplane diff xr1.yaml xr2.yaml --
+
+  # Show the changes that would result from applying xr.yaml (via file) in the 'foobar' namespace with no color output.
+  crossplane diff xr.yaml -n foobar --no-color
+
+  # Show the changes in a compact format with minimal context.
+  crossplane diff xr.yaml --compact
+`
+}
+
+// AfterApply implements kong's AfterApply method to bind our dependencies.
+func (c *Cmd) AfterApply(ctx *kong.Context, log logging.Logger, config *rest.Config) error {
+	return c.initializeDependencies(ctx, log, config)
+}
+
+func (c *Cmd) initializeDependencies(ctx *kong.Context, log logging.Logger, config *rest.Config) error {
+	config = c.initRestConfig(config, log)
+	appCtx, err := NewAppContext(config, log)
+	if err != nil {
+		return errors.Wrap(err, "cannot create app context")
+	}
+	proc := makeDefaultProc(c, appCtx, log)
+
+	loader, err := makeDefaultLoader(c)
+	if err != nil {
+		return errors.Wrap(err, "cannot create resource loader")
+	}
+
+	ctx.Bind(appCtx)
+	ctx.BindTo(proc, (*dp.DiffProcessor)(nil))
+	ctx.BindTo(loader, (*internal.Loader)(nil))
+	return nil
+}
+
+func (c *Cmd) initRestConfig(config *rest.Config, logger logging.Logger) *rest.Config {
+	// Set default QPS and Burst if they are not set in the config
+	// or override with values from options if provided
+	originalQPS := config.QPS
+	originalBurst := config.Burst
+
+	if c.QPS > 0 {
+		config.QPS = c.QPS
+	} else if config.QPS == 0 {
+		config.QPS = 20
+	}
+
+	if c.Burst > 0 {
+		config.Burst = c.Burst
+	} else if config.Burst == 0 {
+		config.Burst = 30
```

Comment:
You may be able to just disable these. The API server is pretty good about letting clients know when they need to backoff these days, so folks are increasingly dropping explicit client-side rate limits. (Read up on API Priority and Fairness for more context.)